week3/homework/src/index.js

```diff
@@ -1,3 +1,88 @@
 'use strict';
 
-// TODO: Write the homework code in this file
+const Express = require('express');
+
+// import our CRUD actions
+const { createTodo, readTodos, updateTodo, deleteTodo } = require('./actions');
+
+const Todo = require('./todo');
+
+const FILENAME = 'todos.json';
+const PORT = 3000;
+const TODO_SLUG = 'todos';
+
+const todo = new Todo(FILENAME);
+
+const app = new Express();
+
+// Use built-in JSON middleware to automatically parse JSON
+app.use(Express.json());
+
+// Functions that I created
+function readTodo(todo, request, response) {
+  todo.read().then(todos => {
+    const todoId = request.params.id;
+    const pickTodo = todos.find(todo => todo.id === todoId);
+    if (pickTodo) {
+      response.send(pickTodo);
+    } else {
+      response.status(404).send({ message: `user with id ${todoId} not found` });
+    }
+  });
+}
+
+function clearTodos(todo, request, response) {
+  todo
+    .delete_()
+    .then(() => {
+      response.status(204);
+      response.end();
+    })
+    .catch(({ message }) => {
+      response.status(500);
+      response.json({ error: message });
+    });
+}
+
+function markAsDone(todo, req, res) {
+  const { id } = req.params;
+  const done = true;
+  todo
+    .updateDone(id, done)
+    .then(todo => {
+      res.send(todo);
+    })
+    .catch(error => {
+      res.status(404);
+      res.send({ error: error.message });
+    });
+}
+
+function markAsNotDone(todo, req, res) {
+  const { id } = req.params;
+  const done = false;
+  todo
+    .updateDone(id, done)
+    .then(todo => {
+      res.send(todo);
+    })
+    .catch(error => {
+      res.status(404);
```

Comment:
is this status code always correct? What if for example a file operation fails for some reason?